### PR TITLE
Restructure compile folder

### PR DESCRIPTION
**Pull request by Hofer-Julian**
After a couple of iterations, this now seems to work.

### CODE REVIEW

File: compile/create_binaries/Project.toml [hunk 1]
```diff
@@ -1,2 +1,3 @@
 [deps]
 PackageCompiler = "9b87118b-4619-50d2-8e1e-99f35a4d4d9d"
+Ribasim = "aac5e3d9-0b8f-4d4f-8241-b1a7a9632635"
```
Comment (visr) [inline compile/create_binaries/Project.toml:3]: I think you found that the Ribasim dependency was not needed here, could you remove it and also commit the updated manifest after you do?

Comment (visr) [reply compile/create_binaries/Project.toml:3]: Would be good to also add TOML and LibGit2 deps explicitly (it works since they are stdlibs).

**Review by visr — CHANGES_REQUESTED** [review]
Thanks! Left some small comments. Also in the `libribasim/Project.toml`, could you add

```toml
[compat]
BasicModelInterface = "0.1"
julia = "1.8"

```

File: compile/create_binaries/Project.toml (hunk 1 above)
Comment (Hofer-Julian) [reply compile/create_binaries/Project.toml:3]: Thanks.
I've changed it locally for now.
If the output is valid, I will update this branch.

Comment (Hofer-Julian) [reply compile/create_binaries/Project.toml]: Done

**Review by visr — APPROVED** [review]

**Pull request merged**